packages/core/src/extension/extension-service.ts

```diff
@@ -33,23 +33,19 @@ export default class ExtensionService {
         this._extensionId = extensionId;
 
         this._eventListener.listen();
+        this._commandPoster.setContext({ extensionId });
+        this._internalCommandPoster.post({
+            type: ExtensionInternalCommandType.ResizeIframe,
+            payload: { extensionId },
+        });
     }
 
-    post(event: ExtensionCommand): void {
-        if (!this._extensionId) {
-            return;
-        }
-
-        if (!Object.values(ExtensionCommandType).includes(event.type)) {
-            throw new Error(`${event.type} is not supported.`);
+    post(command: ExtensionCommand): void {
+        if (!Object.values(ExtensionCommandType).includes(command.type)) {
+            throw new Error(`${command.type} is not supported.`);
         }
 
-        const payload = {
-            ...event.payload,
-            extensionId: this._extensionId,
-        };
-
-        this._eventPoster.post({ ...event, payload });
+        this._commandPoster.post(command);
```

Comment:
Not sure if I missed another change, does the client need to pass the extensionId explicitly?

Reply:
Or they don't need to pass extensionId at all now?

Reply:
I think as clients you don't need to pass it, see https://github.com/bigcommerce/checkout-extension-example/blob/4943cf88922f96174fa05fece849032f502eeb65/src/commands/loading-indicator.js#L2-L5

Reply:
Yes, that's correct. We were adding that in the service in order to keep the client side simple.  Ahh I see we now set it as extension context, is that correct?